libqtile/core/manager.py

```diff
@@ -326,39 +329,66 @@ async def finalize(self):
             logger.exception('exception during finalize')
             self.core.remove_listener()
 
-    def _process_fake_screens(self):
+    def _process_fake_screens(self) -> List[Screen]:
         """
         Since Xephyr and Xnest don't really support offset screens, we'll fake
         it here for testing, (or if you want to partition a physical monitor
         into separate screens)
         """
+        screens = []
+
         for i, s in enumerate(self.config.fake_screens):
-            # should have x,y, width and height set
+            # should have x, y, width and height set
             s._configure(self, i, s.x, s.y, s.width, s.height, self.groups[i])
             if not self.current_screen:
                 self.current_screen = s
-            self.screens.append(s)
+            screens.append(s)
 
-    def _process_screens(self) -> None:
+        return screens
+
+    def _process_screens(self) -> List[Screen]:
         if hasattr(self.config, 'fake_screens'):
-            self._process_fake_screens()
-            return
+            return self._process_fake_screens()
 
+        screens = []
         screen_info = self.core.get_screen_info()
 
         for i, (x, y, w, h) in enumerate(screen_info):
-            if i + 1 > len(self.config.screens):
-                scr = Screen()
-            else:
+            if i < len(self.screens):
+                scr = self.screens[i]
+            elif i < len(self.config.screens):
```

Comment:
I feel this might be prone to bugs, wouldn't it be safer to use the screen names instead?

Reply:
What bug risk do you mean? And screen names?

Reply:
The `len(self.screens)` test might be redundant.

Reply:
I'm worried that we use the existing screens on one side of the condition, and config screens on the other side. My brain is quite dead at this time of the day so please bear with me if I'm saying anything dumb, but I have the impression the user change the order of the screens in its configuration, we may end up in a situation where a same Screen object is used for two different physical screens.

Reply:
Got it. Using `self.screens` here was redundant anyway, as the same objects are available in the config. Removing it also will prevent that issue you describe :)